src/main/java/io/confluent/connect/elasticsearch/EnumRecommender.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.connect.elasticsearch;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
+
+@SuppressWarnings("rawtypes")
+class EnumRecommender<T extends Enum> implements ConfigDef.Validator, ConfigDef.Recommender {
+
+  private final Set<String> validValues;
+  private final Class<T> enumClass;
+
+  public EnumRecommender(Class<T> enumClass) {
+    this.enumClass = enumClass;
+    Set<String> validEnums = new LinkedHashSet<>();
+    for (Object o : enumClass.getEnumConstants()) {
+      String key = o.toString().toLowerCase();
+      validEnums.add(key);
+    }
+
+    this.validValues = Collections.unmodifiableSet(validEnums);
+  }
+
+  @Override
+  public void ensureValid(String key, Object value) {
+    // calling toString on itself because IDE complains if the Object is passed.
+    if (value != null && !validValues.contains(value.toString())) {
+      throw new ConfigException(key, value, "Invalid enumerator");
+    }
```

Comment:
```suggestion
    String enum = value.toString().toUpperCase();
    if (value != null && !validValues.contains(enum)) {
      throw new ConfigException(key, value, "Value must be one of: " + toString());
    }
```

wanna be case insensitive and be more descriptive with the error message